enhancements/sig-architecture/6-placements/README.md

```diff
@@ -0,0 +1,523 @@
+	// RequiredDuringSchedulingRequiredDuringExecution is a scheduling strategies for
+	// ClusterAntiAffinity. The anti-affinity requirements specified by this field should be met
+	// at both scheduling time (when the placement decisions are made for the first time) and
+	// execution time (After one or multiple placement decisions are made). It means,
+	// 1) Any cluster, which does not meet the requirements, should not be selected;
+	// 2) If a selected cluster ceases to meet the requirements (e.g. due to an update) after the
+	//    placement decisions are made, it will be eventually removed from the placement decisions;
+	// 3) If a cluster starts to meet the requirements after the placement decisions are made, it
+	//    will either be selected or at least has chance to be selected;
+	// When there are multiple elements, the lists of clusters corresponding to each
+	// ClusterAffinityTerm are intersected, i.e. all terms must be satisfied.
+	//
+	// Other scheduling strategies not yet implemented includes
+  // RequiredDuringSchedulingIgnoredDuringExecution and PreferredDuringSchedulingIgnoredDuringExecution
+	// +optional
+	RequiredDuringSchedulingRequiredDuringExecution []ClusterAffinityTerm `json:"requiredDuringSchedulingRequiredDuringExecution,omitempty"`
+}
+
+// The selected clusters could not co-located with each other,
+// where co-located is defined as the value of the label/claim with name <topologyKey> is the same.
+type ClusterAffinityTerm struct {
+	// TopologyKey is either a label key or a cluster claim name of clusters
+	// +required
+	TopologyKey string `json:"topologyKey"`
+
+	// TopologyKeyType indicates the type of TopologyKey. It could be Label or Claim.
+	// +required
+	TopologyKeyType string `json:"topologyKeyType"`
+}
+
+const (
+	TopologyKeyTypeLabel string = "Label"
+	TopologyKeyTypeClaim string = "Claim"
+)
+
+type PlacementStatus struct {
+	// NumberOfSelectedClusters represents the number of selected clusters
+	NumberOfSelectedClusters int32 `json:"numberOfSelectedClusters"`
+
+	// Conditions contains the different condition statuses for this Placement.
+	Conditions []metav1.Condition `json:"conditions"`
+}
+
+const (
+	// PlacementConditionSatisfied means Placement is satisfied.
+	// A placement is satisfied only if all predicates are satisfied. And a predicate
+	// is not satisfied if
+	// 1) NumberOfClusters is specified;
+	// 2) And the number of matched clusters for this predicate is less than NumberOfClusters;
+	PlacementConditionSatisfied string = "PlacementSatisfied"
+)
+```
+
+#### PlacementDecision API
+`PlacementDecision` represents a slice of clusters selected by a placement. A `Placement` can link to multiple `PlacementDecisions`, and uses a label selector for reference.
+```go
+// PlacementDecision indicates a decision from a placement
+// PlacementDecision should has a label placement.open-cluster-management.io={placement name}
+// to reference a certain placement.
+type PlacementDecision struct {
```

Comment:
I think you'll want an example of how a controller could handle a cluster being removed from a PlacementDecision that happens while the consuming controller is off.  I think the API is fine, but an explanation and/or sample code for handling the inverted match will help avoid some mistakes.

Reply:
An example is added as suggested.